lambda-functions/receiveNotificationRequest/src/main/java/com/paladincloud/SendNotification.java

```diff
@@ -0,0 +1,56 @@
+package com.paladincloud;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.LambdaLogger;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.sns.AmazonSNS;
+import com.amazonaws.services.sns.AmazonSNSClientBuilder;
+import com.amazonaws.services.sns.model.PublishRequest;
+import com.amazonaws.services.sns.model.PublishResult;
+import com.google.gson.*;
+
+import java.util.List;
+import java.util.Map;
+
+
+public class SendNotification implements RequestHandler<Map<String,Object>, String> {
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+    /**
+     *Below method places all incoming notification requests into topic which are consumed
+     * by subsequent lambda functions.
+     *
+     */
+    @Override
+    public String handleRequest(Map<String,Object> event, Context context)
+    {
+        LambdaLogger logger = context.getLogger();
+        String response = new String("hello paladin cloud");
+        logger.log("EVENT: " + gson.toJson(event));
+        AmazonSNS client = AmazonSNSClientBuilder.standard().build();
+        try {
+            String snsTopicArn = System.getenv("SNS_TOPIC_ARN");
+            String payloadString = (String)event.get("body");
```

Comment:
on this line here, we could add a check that states:

if "body" or "Message", where "Message" object is the payload contents when the trigger to lambda is from sns

Reply:
if trigger is from sns, below if block (at line number 34) will take care...

Sample payload from sns

{
    "Records": [
        {
            "EventSource": "aws:sns",
            "EventVersion": "1.0",
            "EventSubscriptionArn": "arn:aws:sns:us-east-1:765245176332:test-paladin:584583da-263b-4215-82af-82b000d5be03",
            "Sns": {
                "Type": "Notification",
                "MessageId": "a6b59f09-1ffa-5370-9432-c3a44bdf925d",
                "TopicArn": "arn:aws:sns:us-east-1:765245176332:test-paladin",
                "Message": "{\n  \"eventId\": \"820dcb5e-c02e-40ee-ac53-0d3b6f3cdffc\",\n  \"eventName\": \"Violation for policy - Check the Expiry status of the ACM Certificate\",\n  \"eventCategory\": \"violations\",\n  \"eventCategoryName\": \"Violations\",\n  \"eventSource\": \"paladinCloud\",\n  \"eventSourceName\": \"PaladinCloud\",\n  \"eventDescription\": \"Violation for policy - Check the Expiry status of the ACM Certificate\",\n  \"subject\": \"Policy Violation Created\",\n  \"payload\": {\n    \"resourceId\": \"*.paladincloud.io\",\n    \"resourceIdLink\": \"https://dev.paladincloud.io/pl/assets/asset-list/acmcertificate/*.paladincloud.io?ag\\u003daws\",\n    \"issueId\": \"795da2f94f588845b32a6762d2ea13a0\",\n    \"issueIdLink\": \"https://dev.paladincloud.io/pl/compliance/issue-listing/issue-details/795da2f94f588845b32a6762d2ea13a0?ag\\u003daws\",\n    \"policyName\": \"Check the Expiry status of the ACM Certificate\",\n    \"policyNameLink\": \"https://dev.paladincloud.io/pl/compliance/policy-knowledgebase-details/ACMCertificate_Expiry_For_X_Days_version-1_ACMCertificate_Expiry_acmcertificate/true?ag\\u003daws\",\n    \"description\": \"SSL(ACM) Expiry within 90 days found!!\",\n    \"scanTime\": \"2023-04-10T00:01:27.794Z\",\n    \"action\": \"create\"\n  }\n}",
                "Timestamp": "2023-04-10T00:01:41.604Z",
                "SignatureVersion": "1",
                "Signature": "MzfcLkqRIBCfFIFZi64tWROiUxhHwo+4KJT1nCeEz6TEb4q6qWXKVPYICLNfMpvC+Y+DvV7IGSWReEEO+ge+Eowsibf8dHrHWrQ5kWkwlVTz9Fg9xc6GznpS+fdQACxS9UZT1pYkLRx5vGLGEt3SfasX0bnFzTUv9rSKhh1VM44mFC+yb3V5oDFdShhpPnXt88yrr6tMGlD/9Y+ttSKhiv87gRVqAha336pjVYzhv4vYyGpvfbrcw2YNVWtHlaPUAw1y/w8v2Jbyahk3neBdpPRmEPcHB+0QpC8dzKaxIajLGNFSb71j7HrPAN7phaYhQWVBs0nCvoO8r22nEJapYA==",
                "SigningCertUrl": "https://sns.us-east-1.amazonaws.com/SimpleNotificationService-56e67fcb41f6fec09b0196692625d385.pem",
                "UnsubscribeUrl": "https://sns.us-east-1.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:us-east-1:765245176332:test-paladin:584583da-263b-4215-82af-82b000d5be03",
                "MessageAttributes": {}
            }
        }
    ]
}